services/MerchStoreService.ts

```diff
@@ -431,34 +441,236 @@ export default class MerchStoreService {
-    }
-    await this.transactions.readWrite(async (txn) => {
-      const orderItemRepository = Repositories.merchOrderItem(txn);
-      const orderItems = await orderItemRepository.batchFindByUuid(Array.from(fulfillmentUpdates.map((oi) => oi.uuid)));
-      if (orderItems.size !== fulfillmentUpdates.length) {
-        throw new NotFoundError('Missing some order items');
+  private static isLessThanTwoDaysBeforePickupEvent(pickupEvent: OrderPickupEventModel): boolean {
+    return new Date() > moment(pickupEvent.start).subtract(2, 'days').toDate();
+  }
+
+  public async editMerchOrderPickup(orderUuid: Uuid, pickupEventUuid: Uuid, user: UserModel): Promise<OrderModel> {
+    return this.transactions.readWrite(async (txn) => {
+      const orderRepository = Repositories.merchOrder(txn);
+      const order = await orderRepository.findByUuid(orderUuid);
+      if (!order) throw new NotFoundError('Order not found');
+      if (order.user.uuid !== user.uuid) throw new ForbiddenError('Cannot edit the order of a different user');
+      if (MerchStoreService.isInactiveOrder(order)) throw new UserError('Cannot modify pickup for inactive orders');
+
+      const pickupEvent = await Repositories.merchOrderPickupEvent(txn).findByUuid(pickupEventUuid);
+      if (!pickupEvent) throw new NotFoundError('Order pickup event not found');
+      if (MerchStoreService.isLessThanTwoDaysBeforePickupEvent(pickupEvent)) {
+        throw new UserError('Cannot change order pickup to an event that starts in less than 2 days');
+      }
+      const orderInfo = await MerchStoreService.buildOrderUpdateInfo(order, pickupEvent, txn);
+      await this.emailService.sendOrderPickupUpdated(user.email, user.firstName, orderInfo);
+      return orderRepository.upsertMerchOrder(order, { pickupEvent });
+    });
+  }
+
+  private static isInactiveOrder(order: OrderModel): boolean {
+    return order.status === OrderStatus.FULFILLED || order.status === OrderStatus.CANCELLED;
+  }
+
+  /**
+   * Marks an order as missed. An order can be marked as missed only if it's previous status was PLACED,
+   * and its associated pickup event has already passed. Only admins can mark orders as missed
+   * @param uuid order uuid
+   * @returns updated order
+   */
+  public async markOrderAsMissed(uuid: Uuid, proxy: UserModel): Promise<OrderModel> {
+    return this.transactions.readWrite(async (txn) => {
+      const orderRespository = Repositories.merchOrder(txn);
+      const order = await orderRespository.findByUuid(uuid);
+      if (!order) throw new NotFoundError('Order not found');
+      if (order.status !== OrderStatus.PLACED) {
+        throw new UserError('Cannot mark an order as missed if it\'s already been cancelled, missed, or fulfilled');
+      }
+      // compare with start date and not end date so that store admins can mark orders
+      // as missed during the event and not necessarily limited to after the event
+      // (e.g. in the case where the event ends early)
```

Comment:
was a little confused by the wording, I think this is saying the same thing?
```suggestion
      // as missed during and after the event
```